src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Avatar from '@material-ui/core/Avatar';
+import { Button } from '@material-ui/core';
+import './Dropdown.scss';
+
+interface Dropdownprops {
+  DropdownOpener: React.ReactNode;
+}
+
+export const Dropdown: React.FC<Dropdownprops> = ({ DropdownOpener }) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState<boolean>(false);
+  const box = useRef(null);
```

Comment:
```useRef() as MutableRefObject<HTMLDivElement>``` works wonders with typescript

Reply:
You won't need to check for nulls (Only in useEffect I think, but it works)

Reply:
const box =  useRef(null) as MutableRefObject<HTMLDivElement>